frappe/desk/reportview.py

```diff
@@ -18,20 +19,123 @@
 @frappe.read_only()
 def get():
 	args = get_form_params()
+	return compress(execute(**args), args=args)
 
-	data = compress(execute(**args), args = args)
-
-	return data
+@frappe.whitelist()
+@frappe.read_only()
+def get_list():
+	# uncompressed (refactored from frappe.model.db_query.get_list)
+	return execute(**get_form_params())
 
 def execute(doctype, *args, **kwargs):
 	return DatabaseQuery(doctype).execute(*args, **kwargs)
 
 def get_form_params():
 	"""Stringify GET request parameters."""
 	data = frappe._dict(frappe.local.form_dict)
+	clean_params(data)
+	parse_json(data)
+
+	validate_fields(data)
+	if data.filters:
+		validate_filters(data, data.filters)
+	if data.or_filters:
+		validate_filters(data, data.or_filters)
+
+	data.strict = None
+
+	return data
+
+def get_fields(data):
+	if ((isinstance(fields, string_types) and fields == "*")
+		or (isinstance(fields, (list, tuple)) and len(fields) == 1 and fields[0] == "*")):
+		data["fields"] = frappe.db.get_table_columns(data.doctype)
+		fields = data["fields"]
+
+def validate_fields(data):
+	update_star_field_param(data)
+
+	for field in data.fields:
+		fieldname = extract_fieldname(field)
+		if is_standard(fieldname): continue
```

Comment:
We need to also continue if the extracted value is `"*"`. Currently, `count(*)` does not work.
Otherwise, will need to replace existing code with something like `count(name)`.